### ISSUE TITLE
Error: Exception in ASGI Application

### ISSUE THREAD

**Issue by daxmonson** [debanjum/khoj#139]
entered the below string in Terminal. Then opened khoj in Obsidian
pip install khoj-assistant && khoj --no-gui

Terminal downloaded a bunch files and got to special_tokens_map.json: 100%

Then got this error message:

"GET /api/update?t=markdown HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application

Any ideas here? Thanks

**Comment by debanjum** [comment]
Thanks for reporting this!

> Then got this error message:
>
> "GET /api/update?t=markdown HTTP/1.1" 500 Internal Server Error
ERROR: Exception in ASGI application

I'll need more details to debug what the potential issue could be:
- Can you share a screenshot (or more details) of what the terminal shows when it throws the ASGI error?
- Also what Operating System and Python version are you using?

**Comment by daxmonson** [comment]
I tried running the install again:  pip install khoj-assistant && khoj --no-gui

This time the ASGI errror didn't come up, but got this: 
<img width="1066" alt="Screen Shot 2023-02-05 at 6 30 22 PM" src="https://user-images.githubusercontent.com/5855930/216869788-9f11085f-49d2-4ad6-9970-53ef0179f29d.png">

Thanks for taking a look.

**Comment by debanjum** [comment]
Thanks for sharing a screenshot of the new issue you're seeing. It seems like the gzipped jsonl files khoj creates to index your markdown notes is corrupted. 

Can you 
1. Delete your `~/.khoj` directory. It should be at `/Users/<your-username>/.khoj`
2. Restart khoj by running the `khoj --no-gui` command again?

**Comment by debanjum** [comment]
Hi,

I'll need more details to debug your issue. Can you start the khoj backend in verbose mode with `khoj -vv` and share what you see in the terminal before/after you try searching?

> Is there an indexing that would be as long as overnight?

No, the indexing shouldn't take more than a few minutes. It takes ~6 minutes to index ~15K notes on a Macbook Pro M1

- While khoj is indexing you should see something like:
   `Batches: 80%  |█████████   | 80/100 [06:20<00:00,  1.30it/s]`
- Once khoj is done indexing you should see something like:
   `Batches: 100%|██████████| 100/100 [06:20<00:00,  1.30it/s]`

> Think I'll wait a few days and see if you can get it installed with the latest Python and see if that helps.

Khoj was tested to work with python 3.10 on a Mac already. 
The fix to make Khoj work with python 3.11 doesn't seem relevant for your issue.

**Comment by daxmonson** [comment]
Hi, 
 Had a dependency that broke downgrading Python, so I'll have to get back it this weekend when there's more time. Set a reminder for that. Thanks again!

**Comment by debanjum** [comment]
Closing this as there was insufficient information to debug or reproduce the reported ASGI Application issue
